graphql/resolvers.go

```diff
@@ -1819,6 +1819,33 @@ func (r *mutationResolver) UnschedulePatchTasks(ctx context.Context, patchID str
 	return &patchID, nil
 }
 
+// We only allow scheduling undispatched base tasks for tasks that are failing on the current patch
+func (r *mutationResolver) ScheduleUndispatchedBaseTasks(ctx context.Context, patchID string) ([]*restModel.APITask, error) {
+	opts := data.TaskFilterOptions{
+		Statuses:     evergreen.TaskFailureStatuses,
+		BaseStatuses: []string{evergreen.TaskUndispatched},
+	}
+	tasks, _, err := r.sc.FindTasksByVersion(patchID, opts)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Could not fetch tasks for patch :%s ", err.Error()))
+	}
+
+	restartedTasks := []*restModel.APITask{}
+	for _, t := range tasks {
+		// Can ignore error while fetching tasks this could just mean the task didn't exist on the base commit
+		baseTask, _ := t.FindTaskOnBaseCommit()
```

Comment:
if the task is a generated task, should we try to find the generated task on the base commit if baseTask == nil?

Reply:
we may also want to reuse this logic https://github.com/evergreen-ci/evergreen/blob/176874716e4e9dccb4f6c1b5ff06e558f83a36e0/model/task_lifecycle.go#L186 so that after the generator task runs we know which tasks we want to schedule explicitly (I wish I'd called it something less specific than Stepback but alas)